server.go

```diff
@@ -1569,11 +1575,41 @@ func (s *server) listenHandler(listener net.Listener) {
 		sp.Peer = peer.NewInboundPeer(newPeerConfig(sp))
 		go s.peerDoneHandler(sp)
 		sp.Connect(conn)
+		addr, err := sp.RemoteAddr()
+		if err == nil {
+			sp.isWhitelisted = s.isWhitelisted(addr)
+		}
 	}
 	s.wg.Done()
 	srvrLog.Tracef("Listener handler done for %s", listener.Addr())
 }
 
+// isWhitelisted returns whether the IP address is included in the whitelisted
+// networks and IPs.
+func (s *server) isWhitelisted(addr net.Addr) bool {
```

Comment:
Any reason this is defined on the server type as opposed to being a standalone function?  The code doesn't appear to use anything from the server instance.